homeassistant/components/homekit_controller/fan.py

```diff
@@ -139,23 +188,81 @@ async def async_turn_off(self, **kwargs: Any) -> None:
         await self.async_put_characteristics({self.on_characteristic: False})
 
 
-class HomeKitFanV1(BaseHomeKitFan):
+class HomeKitFanV1(HomeKitFan):
     """Implement fan support for public.hap.service.fan."""
 
     on_characteristic = CharacteristicsTypes.ON
 
 
-class HomeKitFanV2(BaseHomeKitFan):
+class HomeKitFanV2(HomeKitFan):
     """Implement fan support for public.hap.service.fanv2."""
 
     on_characteristic = CharacteristicsTypes.ACTIVE
 
 
+class EcobeeFan(BaseHomeKitFan):
+    """Implement fan support for ecobee fans."""
+
+    speed_characteristic = CharacteristicsTypes.VENDOR_ECOBEE_FAN_WRITE_SPEED
+    default_percentage = 50
+
+    @property
+    def percentage(self) -> int:
+        """Return the current speed percentage."""
+        return ranged_value_to_percentage(
+            self._speed_range, self.service.value(self.speed_characteristic)
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if device is on."""
+        return bool(self.percentage)
+
+    async def async_set_percentage(self, percentage: int) -> None:
+        """Set the speed of the fan."""
+        # Sending the fan mode request sometimes ends up getting ignored by ecobee
+        # and this might be because it the older value instead of newer, and ecobee
+        # thinks there is nothing to do.
+        # So in order to make sure that the request is executed by ecobee, we need
+        # to send a different value before sending the target value.
+        # Fan mode value is a value from 0 to 100. We send a value off by 1 first.
+        value = ranged_value_to_percentage(self._speed_range, percentage)
+        if value > self._min_speed:
+            other_value = value - 1
+        else:
+            other_value = self._min_speed + 1
+
+        if value != other_value:
+            await self.async_put_characteristics(
+                {self.speed_characteristic: other_value}
+            )
+        await self.async_put_characteristics({self.speed_characteristic: value})
+
+    async def async_turn_on(
+        self,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Turn the specified fan on."""
+        await self.async_set_percentage(percentage or self.default_percentage)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the specified fan off."""
+        await self.async_set_percentage(0)
```

Comment:
I have no idea what this actually does.  I'm tempted to take it back out

Reply:
So no matter what I set this value to, it holds it for a bit and then resets to 0 after 2-3m

I'm not sure it actually does anything

Reply:
Setting this to any value will change the device fan mode On to Auto but I don't know what it actually does

Reply:
Setting it to 100 will set it to On
Setting it to anything less than 100 will set it to Auto

Reply:
But target fan mode does the same thing

Reply:
https://github.com/home-assistant/core/pull/60914#issue-1070618559

Well the original PR says that 

We should remove this and just implement Target Fan Mode